packages/core/strapi/bin/strapi.js

```diff
@@ -118,68 +118,20 @@ program
   .description('Start your Strapi application in development mode')
   .action(getLocalScript('develop'));
 
-// `$ strapi generate:api`
+// $ strapi generate
 program
-  .command('generate:api <id> [attributes...]')
-  .option('-a, --api <api>', 'API name to generate the files in')
-  .option('-p, --plugin <api>', 'Name of the local plugin')
-  .option('-e, --extend <api>', 'Name of the plugin to extend')
-  .option('--draft-and-publish', 'Enable draft/publish', false)
-  .description('Generate a basic API')
-  .action((id, attributes, cliArguments) => {
-    cliArguments.attributes = attributes;
-    getLocalScript('generate')(id, cliArguments);
-  });
-
-// `$ strapi generate:controller`
-program
-  .command('generate:controller <id>')
-  .option('-a, --api <api>', 'API name to generate the files in')
-  .option('-p, --plugin <api>', 'Name of the local plugin')
-  .option('-e, --extend <api>', 'Name of the plugin to extend')
-  .description('Generate a controller for an API')
-  .action(getLocalScript('generate'));
-
-// `$ strapi generate:model`
-program
-  .command('generate:model <id> [attributes...]')
-  .option('-a, --api <api>', 'API name to generate a sub API')
-  .option('-p, --plugin <api>', 'plugin name')
-  .option('--draft-and-publish', 'Enable draft/publish', false)
-  .description('Generate a model for an API')
-  .action((id, attributes, cliArguments) => {
-    cliArguments.attributes = attributes;
-    getLocalScript('generate')(id, cliArguments);
+  .command('generate')
+  .description('Launch interactive API generator')
+  .action(() => {
+    require('@strapi/generate')();
```

Comment:
Before calling the generate we should at least check `checkCwdIsStrapiApp(name)` so you don't run it from everywhere for now